solutions/fully-configurable/main.tf

```diff
@@ -191,6 +191,13 @@ locals {
       additional_security_group_ids = pool.additional_security_group_ids
       subnet_prefix                 = "default"
   } if length(pool.vpc_subnets) == 0])
+
+  # Managing the ODF version accordingly, as it changes with each OCP version.
+  addons = lookup(var.addons, "openshift-data-foundation", null) != null ? lookup(var.addons["openshift-data-foundation"], "version", null) == null ? { for key, value in var.addons :
```

Comment:
Just checking that this was tested when fully upgrading when user had supplied addons previously? You are basically rewriting the whole addons block here just to modify the one value, and you will be potentially adding additional fields that may not have been there before, and I was curious if this causes terraform to think it needs to reinstall previous addons?

For example:
user had previously installed debug tool by having variable
```
{
  debug-tool = {}
}
```

This routine will change this slightly to
```
{
  debug-tool = { 
    version = null
    parameters_json = null
  }
}
```

Is there a better way to do this without rewriting the entire map? for instance maybe you could do a merge of just the one addon changes instead?